src/perf/lib/Tcp.h

```diff
@@ -249,7 +249,7 @@ class TcpConnection {
     QUIC_BUFFER* NewSendBuffer();
     void FreeSendBuffer(QUIC_BUFFER* SendBuffer);
     bool FinalizeSendBuffer(QUIC_BUFFER* SendBuffer);
-    void AddRef() { CxPlatRefIncrement(&Ref); }
+    bool TryAddRef() { return CxPlatRefIncrementNonZero(&Ref, 1) != FALSE; }
```

Comment:
>CxPlatRefIncrementNonZero [](http://example.com/codeflow?start=30&length=25)

Is the purpose of TryAddRef return false if ref == 0?

Reply:
Yes.

Reply:
But if Ref drops to zero, the entire connection object has been already deleted?

Reply:
It is _being_ deleted. In the destructor, we delete the socket, which has to wait for any outstanding receives. That's the race here.

Reply:
Oh, I got it.